Pre-processing scripts to create graph genomes, correct allele bias mapping, and count reads to each allele are found here.

Create Graph genome:
* Building_vg_genomes.sh
* Building_fasta_variant_sequences.sh
* Construct_fasta_variant_sequences.sh

Map to graph genome:
* DonorSpecificMapping

Comparison to bowtie2:
* PatientSpecificMappingAndBinomTest_bowtie2.sh

WASP filter:
* WASP_correction_vg.sh
* intersect_v2.sh

Count reads:
* Count_Reads_overlapping-allele.sh

Beta-binomial test for ASB:
* Beta-binomial_test.R

VEP annotation:
* vep108.ini
